auth/src/main/java/com/firebase/ui/auth/provider/TwitterProvider.java

```diff
@@ -1,137 +0,0 @@
-package com.firebase.ui.auth.provider;
-
-import android.app.Activity;
-import android.content.Context;
-import android.content.Intent;
-import android.net.Uri;
-import android.support.annotation.LayoutRes;
-import android.util.Log;
-
-import com.firebase.ui.auth.IdpResponse;
-import com.firebase.ui.auth.R;
-import com.google.firebase.auth.AuthCredential;
-import com.google.firebase.auth.TwitterAuthProvider;
-import com.twitter.sdk.android.core.Callback;
-import com.twitter.sdk.android.core.Result;
-import com.twitter.sdk.android.core.Twitter;
-import com.twitter.sdk.android.core.TwitterAuthConfig;
-import com.twitter.sdk.android.core.TwitterConfig;
-import com.twitter.sdk.android.core.TwitterCore;
-import com.twitter.sdk.android.core.TwitterException;
-import com.twitter.sdk.android.core.TwitterSession;
-import com.twitter.sdk.android.core.identity.TwitterAuthClient;
-import com.twitter.sdk.android.core.models.User;
-
-public class TwitterProvider extends Callback<TwitterSession> implements IdpProvider {
-    private static final String TAG = "TwitterProvider";
-
-    private IdpCallback mCallbackObject;
-    private TwitterAuthClient mTwitterAuthClient;
-
-    public TwitterProvider(Context context) {
-        initialize(context);
-        mTwitterAuthClient = new TwitterAuthClient();
-    }
-
-    public static AuthCredential createAuthCredential(IdpResponse response) {
-        if (!response.getProviderType().equalsIgnoreCase(TwitterAuthProvider.PROVIDER_ID)) {
-            return null;
-        }
-        return TwitterAuthProvider.getCredential(response.getIdpToken(), response.getIdpSecret());
-    }
-
-    private static void initialize(Context context) {
```

Comment:
I now see that the `initialize` bit I complained about is stolen from old code.  So I won't block this PR on that, but would be cool if we found a simple improvement!

Reply:
https://github.com/SUPERCILEX/FirebaseUI-Android/commit/e79388fdfa7b43458cd43da6dbecfa41c095e4f8 makes it a little nicer, what do you think?